opentreemap/importer/models/trees.py

```diff
@@ -250,30 +250,59 @@ def validate_otm_id(self):
         return True
 
     def validate_proximity(self, point):
-        plot_ids_from_this_import = TreeImportRow.objects\
-            .filter(import_event=self.import_event)\
-            .filter(plot__isnull=False)\
-            .values_list('plot__pk', flat=True)
+        """Return False if the point is within 10ft of an existing
+        plot, excluding plots that are part of the same import event.
 
-        nearby = Plot.objects\
-                     .filter(instance=self.import_event.instance)\
-                     .filter(geom__distance_lte=(point, D(ft=10.0)))\
-                     .distance(point)\
-                     .exclude(pk__in=plot_ids_from_this_import)\
+        This is written with as a raw SQL query to take advantage of
+        the PostGIS <-> operator which provides indexed nearest
+        neighbor searching.
 
-        oid = self.cleaned.get(fields.trees.OPENTREEMAP_PLOT_ID, None)
-        if oid:
-            nearby = nearby.exclude(pk=oid)
+        The `AND NOT IN` clause skips plots created by the same import
+        event, assuming that your import file is correct, even though
+        trees are close together.
```

Comment:
Sorry, I was confused.

But since this clause excludes plot ID's in the import I think it is also handling the case of updating an existing plot. Which means we could use `LIMIT 1` instead of `LIMIT 2`. Which should mean the only slow edge case is an instance with 0 trees? If so we could start the import with an `exists()` query on `Plot` and use it to skip proximity checks.